packages/components/src/segmented-control/component.tsx

```diff
@@ -38,8 +45,9 @@ function SegmentedControl(
 		baseId,
 		isAdaptiveWidth = false,
 		isBlock = false,
-		id,
+		id: idProp,
```

Comment:
Why do we need an id prop for the SegmentedControl component. In most `*Control` component it's just an internal thing.

Reply:
Just left a similar [comment](https://github.com/WordPress/gutenberg/pull/34017#discussion_r687585337)